src/chainstate/burn/operations/leader_block_commit.rs

```diff
@@ -294,7 +355,59 @@ impl BlockstackOperation for LeaderBlockCommitOp {
             return Err(op_error::BlockCommitBadInput);
         }
-        
+
+        /////////////////////////////////////////////////////////////////////////////////////
+        // This tx must have the expected commit or burn outputs:
+        //    * if there is a known anchor block for the current reward cycle, and this
+        //       block commit descends from that block 
+        //       the commit outputs must = the expected set of commit outputs
+        //    * otherwise, there must be no block commits
+        /////////////////////////////////////////////////////////////////////////////////////
+        if let Some(reward_set_info) = reward_set_info {
+            // we do some check-inversion here so that we check the commit_outs _before_
+            //   we check whether or not the block is descended from the anchor.
+            // we do this because the descended_from check isn't particularly cheap, so
+            //   we want to make sure that any TX that forces us to perform the check
+            //   has either burned BTC or sent BTC to the PoX recipients
+            let expect_pox_descendant = if self.commit_outs.len() == 0 {
+                false
+            } else {
+                if self.commit_outs.len() != reward_set_info.recipients.len() {
+                    warn!("Invalid block commit: expected {} PoX transfers, but commit has {}",
+                          reward_set_info.recipients.len(), self.commit_outs.len());
+                    return Err(op_error::BlockCommitBadOutputs)
+                }
+                for (expected_commit, _) in reward_set_info.recipients.iter() {
+                    if !self.commit_outs.contains(expected_commit) {
+                        warn!("Invalid block commit: expected to send funds to {}, but that address is not in the committed output set",
+                              expected_commit);
+                        return Err(op_error::BlockCommitBadOutputs)
+                    }
+                }
+                true
+            };
+
+            let descended_from_anchor = tx.descended_from(parent_block_height, &reward_set_info.anchor_block)
+                .map_err(|e| {
+                    error!("Failed to check whether parent (height={}) is descendent of anchor block={}: {}",
+                           parent_block_height, &reward_set_info.anchor_block, e);
+                    op_error::BlockCommitAnchorCheck})?;
+            if descended_from_anchor != expect_pox_descendant {
+                if descended_from_anchor {
+                    warn!("Invalid block commit: descended from PoX anchor, but used burn outputs");
+                } else {
+                    warn!("Invalid block commit: not descended from PoX anchor, but used PoX outputs");
+                }
+                return Err(op_error::BlockCommitBadOutputs)
+            }
+        } else {
+            // no recipient info for this sortition, so expect all burns
```

Comment:
so expect ~~all burns~~  **1** burn output (correct?)

Reply:
Yes -- or really just "count" the first burn output. If the transaction includes a bunch of other burn outputs, they aren't counted towards the transaction.